citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java

```diff
@@ -1121,6 +1125,43 @@ protected void doPostJsonResponse(ParameterParser params,
 		return results;
 	}
 
+	protected Map<String, Object> updateIntroduction(ParameterParser params, SessionState state) {
+		Map<String, Object> results = new HashMap<String, Object>();
+		String message;
+		try {
+			String resourceUuid = params.getString("resourceUuid");
+			String resourceId = this.getContentService().resolveUuid(resourceUuid);
+			String introduction = params.get("addSectionHTML");
+			introduction = getFormattedText().processFormattedText(introduction, new StringBuilder(), true, true);
+			introduction = introduction.replaceAll("'", "&apos;");
+			getContentService().addProperty(resourceId, ResourceProperties.PROP_INTRODUCTION, introduction);
+			message = "Resource updated";
+			state.setAttribute(STATE_CITATION_COLLECTION, null);
+		} catch (IdUnusedException e) {
```

Comment:
You should be able to catch SakaiException instead and if you're passing the stack trace you should use logger.warn(String,Throwable) so that the exception is kept separate.